src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/MethodOnTypeBuilderInstantiation.cs

```diff
@@ -79,12 +79,26 @@ public override bool ContainsGenericParameters
         {
             get
             {
-                if (_method.ContainsGenericParameters)
+                if (_method.ContainsGenericParameters || _type.ContainsGenericParameters)
+                {
                     return true;
-                if (!_method.IsGenericMethodDefinition)
-                    throw new NotSupportedException();
+                }
 
-                return _method.ContainsGenericParameters;
+                if (!IsGenericMethod)
+                {
+                    return false;
+                }
```

Comment:
This defers to `_method.IsGenericMethod`, which seems close to `_method.ContainsGenericParameters`. Does the latter inform the former?

Reply:
No, `_method.ContainsGenericParameters` could be true when method is not generic, but the containing type has any open generic parameter.